packages/@ember/-internals/glimmer/lib/helpers/custom.ts

```diff
@@ -62,12 +73,91 @@ function hasDestroyable(manager: HelperManager): manager is HelperManagerWithDes
   return manager.capabilities.hasDestroyable;
 }
 
+let ARGS_CACHES = DEBUG ? new WeakMap<SimpleArgsProxy, Cache<Partial<Arguments>>>() : undefined;
+
+function getArgs(proxy: SimpleArgsProxy): Partial<Arguments> {
+  return getValue(DEBUG ? ARGS_CACHES!.get(proxy)! : proxy.argsCache!)!;
+}
+
+class SimpleArgsProxy {
```

Comment:
Does something like this already exist for modifiers?

Reply:
It's a more complicated version which proxies each individual argument. Using that would require users to pass in each arg as a function, which would be a bit prohibitive and is not what was spec'd. So, this one is a simpler proxy that basically calls the users compute function and then returns the proper value whenever `positional` or `named` is accessed.